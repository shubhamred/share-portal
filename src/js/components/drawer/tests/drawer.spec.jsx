import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Drawer from '../drawer';

configure({ adapter: new Adapter() });

describe('components/drawer', () => {
  it('Renders default options', () => {
    const props = {
      children: <p> test </p>
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(
      <Drawer {...props} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });

  describe('Check for click handle', () => {
    const props = {
      children: <p> test </p>,
      open: false,
      title: 'dummy',
      handleClose: jest.fn()
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const ButtonComponentWrapper = shallow(<Drawer {...props} />);

    it('check prop type children should not be empty', () => {
      expect(ButtonComponentWrapper.props().children).toBeTruthy();
    });

    it('should handle click', () => {
      ButtonComponentWrapper.simulate('click');
    });
  });
});
