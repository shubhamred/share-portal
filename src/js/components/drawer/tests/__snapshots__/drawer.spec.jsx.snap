// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/drawer Renders default options 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  className="makeStyles-drawer-4"
  classes={
    Object {
      "paper": "makeStyles-drawerPaper-5",
    }
  }
  onClose={[Function]}
  open={false}
>
  <div
    className="makeStyles-drawerHeader-6"
  >
    <div
      className="makeStyles-draweLogo-7"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-9"
        noWrap={true}
        variant="h5"
      />
      <WithStyles(ForwardRef(IconButton))
        aria-label="open drawer"
        className="makeStyles-drawerIcon-8"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <Memo(ForwardRef(ChevronLeftIcon))
          style={
            Object {
              "color": "#1518AF",
            }
          }
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </div>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))
    className="makeStyles-listContainer-12"
    style={
      Object {
        "overflowX": "hidden",
        "width": "270px",
      }
    }
  >
    <p>
       test 
    </p>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Drawer))>
`;
